### PR TITLE
feat(docs): update use-on-error documentation

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
added: An explanation of how **refine** internally uses `useOnError`.

### CODE REVIEW

File: documentation/docs/api-reference/core/hooks/auth/useOnError.md [hunk 1]
```diff
@@ -7,14 +7,14 @@ source: /packages/core/src/hooks/auth/useOnError/index.ts
...
 :::caution
-This hook can only be used if the `authProvider` is provided.
+This hook can only be used if the [`authProvider`][auth-provider] is provided.
 :::
 
-`useOnError` calls the `onError` method from the [`authProvider`](/api-reference/core/providers/auth-provider.md) under the hood.
+`useOnError` calls the [`onError`][on-error] method from the [`authProvider`][auth-provider] under the hood.
 
-It returns the result of `react-query`'s [useMutation](https://react-query.tanstack.com/reference/useMutation) which includes many properties, some of which being isSuccess and isError.
+It returns the result of `react-query`'s [useMutation](https://react-query.tanstack.com/reference/useMutation) which includes many properties, some of which being `isSuccess` and `isError`.
```
Comment (DoguhanOzgurAkca) [inline documentation/docs/api-reference/core/hooks/auth/useOnError.md:15]:
```suggestion
It returns the result of `react-query`'s [useMutation](https://react-query.tanstack.com/reference/useMutation), which includes many properties like `isSuccess` and `isError`.
```
A generally shortened version like

`useOnError` calls the [`onError`][on-error] method from the [`authProvider`][auth-provider] under the hood, which returns the result of `react-query`'s [useMutation](https://react-query.tanstack.com/reference/useMutation) that has many properties like `isSuccess` and `isError`.

could also be good for this, i reckon.

File: documentation/docs/api-reference/core/hooks/auth/useOnError.md [hunk 2]
```diff
@@ -7,14 +7,14 @@ source: /packages/core/src/hooks/auth/useOnError/index.ts
...
 
-Data that is resolved from the `onError` will be returned as the `data` in the query result with the following type:
+Data that is resolved from the [`onError`][on-error] will be returned as the `data` in the query result with the following type:
```
Comment (DoguhanOzgurAkca) [inline documentation/docs/api-reference/core/hooks/auth/useOnError.md:17]:
```suggestion
Data that is resolved from the [`onError`][on-error] hook will be returned as the `data` in the query result with the following type:
```
removing "the" and not adding hook would also be okay, though i think this is better.

File: documentation/docs/api-reference/core/hooks/auth/useOnError.md [hunk 3]
```diff
@@ -24,10 +24,49 @@ type OnErrorResponse = {
...
 -   `logout`: If is `true`, `useOnError` calls the `logout` method.
 -   `error`: An Error object representing any errors that may have occurred during the operation.
 
+## Internal Usage
+
+**refine** uses `useOnError` internally in the data hooks to handle errors in a unified way.
+
+[Refer to Data Provider documentation for more information about data hooks. → ](/docs/api-reference/core/providers/data-provider/#supported-hooks)
+
+When an error is thrown by any data hook, the `useOnError` function is triggered with the error object. Afterward, the error object is passed to the [`onError`][on-error] method of the [`authProvider`][auth-provider]. This error object can be utilized to redirect the user to the login page or to log them out.
```
Comment (aliemir) [inline documentation/docs/api-reference/core/hooks/auth/useOnError.md:39]:
```suggestion
When an error is thrown by any data hook, the `useOnError` function is triggered with the error object. Afterward, the error object is passed to the [`onError`][on-error] method of the [`authProvider`][auth-provider]. This error object can be utilized to redirect the user or to log them out.
```

File: documentation/docs/api-reference/core/hooks/auth/useOnError.md [hunk 4]
```diff
@@ -7,14 +7,14 @@ source: /packages/core/src/hooks/auth/useOnError/index.ts
...
 :::caution
-This hook can only be used if the `authProvider` is provided.
+This hook can only be used if the [`authProvider`][auth-provider] is provided.
 :::
 
-`useOnError` calls the `onError` method from the [`authProvider`](/api-reference/core/providers/auth-provider.md) under the hood.
+`useOnError` calls the [`onError`][on-error] method from the [`authProvider`][auth-provider] under the hood.
 
-It returns the result of `react-query`'s [useMutation](https://react-query.tanstack.com/reference/useMutation) which includes many properties, some of which being isSuccess and isError.
+It returns the result of `react-query`'s [useMutation](https://react-query.tanstack.com/reference/useMutation), which includes many properties like `isSuccess` and `isError`.
```
Comment (DoguhanOzgurAkca) [inline documentation/docs/api-reference/core/hooks/auth/useOnError.md:15]: Shouldn't the link be "https://tanstack.com/query/v4/docs/react/reference/useMutation" ? This one just takes you to the start up page

Comment (aliemir) [inline documentation/docs/api-reference/core/hooks/auth/useOnError.md:15]:
```suggestion
It returns the result of `react-query`'s [useMutation](https://tanstack.com/query/v4/docs/react/reference/useMutation), which includes many properties like `isSuccess` and `isError`.
```

**Review by DoguhanOzgurAkca — APPROVED** [review]

**Review by DoguhanOzgurAkca — APPROVED** [review]

**Review by aliemir — APPROVED** [review]

**Pull request merged**